test/java/seedu/addressbook/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,150 @@
+        assertViewErrorInvalidIndex(typicalAddressBook, listWithAllTypicalPersons,
+                listWithAllTypicalPersons.size() + 1);
+    }
+
+    @Test
+    public void execute_personNotInAddressBook_returnsPersonNotInAddressBookMessage() throws Exception {
+        // generate list with person not in addressbook, add to list
+        ReadOnlyPerson someone = new Person(new Name("me"),
+                                            new Phone("123", true),
+                                            new Email("some@hey.go", true),
+                                            new Address("nus", false),
+                                            new UniqueTagList(Collections.emptySet()));
+        List<ReadOnlyPerson> listWithExtraPerson
+                = new ArrayList<ReadOnlyPerson>(listWithAllTypicalPersons);
+        listWithExtraPerson.add(someone);
+
+        // empty addressbook
+        assertViewErrorPersonNotInAddressBook(emptyAddressBook, listWithExtraPerson, 1);
+
+        // non-empty addressbook
+        assertViewErrorPersonNotInAddressBook(typicalAddressBook, listWithExtraPerson,
+                listWithExtraPerson.size());
+    }
+
+    @Test
+    public void execute_validIndex_returnsPersonDetails() {
+        // person with no private information
+        assertViewSuccess(typicalAddressBook, listWithAllTypicalPersons, 1);
+
+        // person with some private information
+        assertViewSuccess(typicalAddressBook, listWithAllTypicalPersons, 2);
+
+        // person with all private information
+        assertViewSuccess(typicalAddressBook, listWithAllTypicalPersons, 4);
+
+        // addressbook has more people than the list
+        // This can happen when view from filtered list caused by some commands(eg. FindCommand)
+        assertViewSuccess(typicalAddressBook, listWithSomeTypicalPersons, 1);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * invalid index.
+     */
+    private void assertViewErrorInvalidIndex(AddressBook addressBook, List<ReadOnlyPerson> relevantPersons,
+                                             int targetVisibleIndex) {
+        assertViewError(addressBook, relevantPersons, targetVisibleIndex, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Asserts that the details of person at specific index cannot be retrieved due to
+     * person not existing in the addressbook.
+     */
+    private void assertViewErrorPersonNotInAddressBook(AddressBook addressBook, List<ReadOnlyPerson> relevantPersons,
+                                                       int targetVisibleIndex) {
+        assertViewError(addressBook, relevantPersons, targetVisibleIndex, Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+    }
+
+    /**
+     * Asserts that both a ViewCommand and a ViewAllCommand can retrieve from
+     * the {@code addressBook} details of the person at the given {@code index} in the given {@code list}.
```

Comment:
Change `{@code index}` to `{@code targetVisibleIndex}`?

Change `{@code list}` to `{@code relevantPersons} list`?